qiskit/circuit/library/n_local/evolved_operator_ansatz.py

```diff
@@ -106,10 +114,14 @@ def evolved_operator_ansatz(
                 f"({len(parameter_prefix)})."
             )
 
-    num_qubits = operators[0].num_qubits
     if remove_identities:
         operators, parameter_prefix = _remove_identities(operators, parameter_prefix)
 
+    # After removing identities, check if we have any operators left
+    if len(operators) == 0:
```

Comment:
Evolving an identity operator corresponds to adding a global phase, this is not an empty circuit.